### PR TITLE
br: deal with mddljob when pitr restores (#35976)

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/35372

Problem Summary: drop table meta only, not delete record and index belonged to table.

### What is changed and how it works?

deal with the mddl job, and insert rows into table mysql.gc_delete_range.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/stream/rewrite_meta_rawkv.go [hunk 1]
```diff
@@ -436,7 +436,7 @@ func (sr *SchemasReplace) RewriteKvEntry(e *kv.Entry, cf string, insertDeleteRan
 	// skip mDDLJob
 
 	if !strings.HasPrefix(string(e.Key), "mDB") {
-		if strings.HasPrefix(string(e.Key), "mDDLJobH") { // mDDLJobHistory
+		if cf == "default_cf" && strings.HasPrefix(string(e.Key), "mDDLJobH") { // mDDLJobHistory
```
Comment (joccau) [inline br/pkg/stream/rewrite_meta_rawkv.go]: https://github.com/pingcap/tidb/blob/87052dc2f864/br/pkg/stream/rewrite_meta_rawkv.go#L34

**Review by joccau — APPROVED** [review]
LGTM

**Review by 3pointer — APPROVED** [review]

**Pull request merged**